client/src/main/java/org/eclipse/hono/client/CommandClient.java

```diff
@@ -106,4 +106,8 @@
      * @see RequestResponseClient#setRequestTimeout(long)
      */
     Future<Void> sendOneWayCommand(String command, String contentType, Buffer data, Map<String, Object> properties);
+
+    Future<Void> sendCommandWithTenantScopedResponse(String command, Buffer data);
```

Comment:
what about *sendAsyncCommand*? We could add a parameter for the *replyId*. That way a client could not end up in the situation that you described, i.e. sending an async command using a random replyId ...

Reply:
I am fine with the naming :). 

If we pass the replyId here what do you suggest to do with clients that are created with specific replyId, just ignore?

Reply:
Yes, we would need to document this behavior in this method's JavaDoc, of course.

Reply:
this method should be removed with the advent of `AsyncCommandClient`, shouldn't it?